sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -234,65 +244,124 @@ public boolean isSnapshot() {
-            .setSourceIfMatch(sourceModifiedAccessConditions.getIfMatch())
-            .setSourceIfNoneMatch(sourceModifiedAccessConditions.getIfNoneMatch());
+        final SourceModifiedAccessConditions sourceConditions = new SourceModifiedAccessConditions()
+            .setSourceIfModifiedSince(sourceModifiedCondition.getIfModifiedSince())
+            .setSourceIfUnmodifiedSince(sourceModifiedCondition.getIfUnmodifiedSince())
+            .setSourceIfMatch(sourceModifiedCondition.getIfMatch())
+            .setSourceIfNoneMatch(sourceModifiedCondition.getIfNoneMatch());
+
+        final AtomicReference<String> copyId = new AtomicReference<>();
+
+        return new Poller<>(Duration.ofSeconds(1),
+            response -> onPoll(response),
+            () -> {
+                final Mono<BlobsStartCopyFromURLResponse> restResponse = withContext(
+                    context -> azureBlobStorage.blobs().startCopyFromURLWithRestResponseAsync(null, null,
+                        sourceUrl, null, metadata, tier, priority, null, sourceConditions,
+                        destinationAccessConditions.getModifiedAccessConditions(),
+                        destinationAccessConditions.getLeaseAccessConditions(), context));
+
+                return postProcessResponse(restResponse)
+                    .map(response -> {
+                        final BlobStartCopyFromURLHeaders headers = response.getDeserializedHeaders();
+                        copyId.set(headers.getCopyId());
+
+                        return new BlobCopyInfo(sourceUrl.toString(), headers.getCopyId(), headers.getCopyStatus(),
+                            headers.getETag(), headers.getLastModified(), headers.getErrorCode());
+                    });
+            },
+            unused -> {
+                final String copyIdentifier = copyId.get();
+
+                if (!ImplUtils.isNullOrEmpty(copyIdentifier)) {
+                    logger.info("Cancelling copy operation for copy: {}", copyIdentifier);
+
+                    Utility.blockWithOptionalTimeout(abortCopyFromURL(copyIdentifier), null);
+                }
+            });
+    }
 
-        return postProcessResponse(this.azureBlobStorage.blobs().startCopyFromURLWithRestResponseAsync(
-            null, null, sourceURL, null, metadata, tier, priority, null, sourceConditions,
-            destAccessConditions.getModifiedAccessConditions(), destAccessConditions.getLeaseAccessConditions(),
-            context))
-            .map(rb -> new SimpleResponse<>(rb, rb.getDeserializedHeaders().getCopyId()));
+    private Mono<PollResponse<BlobCopyInfo>> onPoll(PollResponse<BlobCopyInfo> pollResponse) {
+        if (pollResponse.getStatus() == OperationStatus.SUCCESSFULLY_COMPLETED
+            || pollResponse.getStatus() == OperationStatus.FAILED) {
+            return Mono.just(pollResponse);
+        }
+
+        return getProperties().map(response -> {
```

Comment:
Is there a maximum number of retries for onPoll operations? Based on poller doc, if the getProperties failed, then the poller will continue to poll. We need to stop somehow in this case.

Reply:
Fixed. It'll end polling if it is unable to get the file properties.